### PR TITLE
Web UI: Add ability to add/remove devices in UI

### PR DESCRIPTION
**Pull request by ghost**
# Changelog

This is our second pull request, after #5.
## TL;DR (if the full changelog below is too long):
- Added ability to add and delete devices
- Ractive finally works and handles displaying devices from DB
- Some UI changes
### index.html
- Now asks you to login or signup instead of suddenly redirecting you to the login page
- Asks you to add a device if you haven’t already.
- Can add devices now.
- Can delete devices now.
- Data actually dependent on database instead of hardcoded examples now
- Uses Bootstrap’s list group and column grid instead of the default Web2Py ones.
- Removed sample code for graphs and Chrome notifications
- _**TODO:** Finish the column under “Configuration” and work with editor team to implement procedure manager. Maybe move JS to separate file if it gets too long?_
### layout.html
- Google Charts JS removed (was meant only as a placeholder)
### default.py
- Code added that handles showing, adding, and deleting devices
- _**KNOWN BUG:** Devices for ALL users shown, need to filter to just specific user._
### tables.py
- Added device_icon field for devices as a standard way of remembering what icon for visual representation a device had. Needed by UI team.
- Added default value for user_email field (was optional and no default values till now)
- _**KNOWN BUG:** db.auth_user.email is not the actual email…what is?_
### _main.scss (NEW)
- Loading animation CSS added
### add.html (NEW)
- Can add devices now (however crudely)
### manage.html (NEW)
- Added as a placeholder for now

### CODE REVIEW

File: applications/server/controllers/default.py [hunk 1]
```diff
@@ -1,25 +1,56 @@
...
+        # Would sign_uuid really be needed here? Leaving just in case.
+        return dict(message=T('Please sign in!'), sign_uuid=sign_uuid)
+    else:
+        device_list = db().select(db.devices.ALL)
+        return dict(device_list=device_list, sign_uuid=sign_uuid)
+
+
+@auth.requires_login()
+def add():
+    db.devices.device_id.writable = True
```
Comment (lucadealfaro) [inline applications/server/controllers/default.py:22]: Please write a doc string below each method that explains what it does.

File: applications/server/controllers/default.py [hunk 2]
```diff
@@ -1,25 +1,56 @@
...
+    form = SQLFORM(db.devices)
+    if form.process().accepted:
+        session.flash = "Device added!"
+        redirect(URL('default', 'index'))
+    return dict(form=form)
+
+
+def load_devices():
+    # TODO: Condense "rows" to just for that one specific user instead of ALL devices
+    rows = db(db.devices).select()
```
Comment (lucadealfaro) [inline applications/server/controllers/default.py:33]: Yes, can you integrate here the work done  by the permissions team to display only devices the user has access to? 
This way we can also see if the permissions are well designed..

File: applications/server/controllers/default.py [hunk 3]
```diff
@@ -1,25 +1,56 @@
...
+                       'last_sync': r.last_sync,
+                       'user_email': r.user_email,
+                       'id': r.id}
+         for r in rows}
+    return response.json(dict(device_dict=d))
+
+
+@auth.requires_signature()
+def delete_devices():
+    delete_devices = request.vars.get("delete_devices[]")
```
Comment (lucadealfaro) [inline applications/server/controllers/default.py:47]: Please add check that the user has the access to delete the device.

File: applications/server/controllers/default.py [hunk 4]
```diff
@@ -1,25 +1,56 @@
...
+    if type(delete_devices) is str:
```
Comment (lucadealfaro) [inline applications/server/controllers/default.py:48]: I think this could be isinstance(delete_devices, basestring) or something like that.

**Comment by lucadealfaro** [comment]
I am going to merge this pull request, but please take care of the couple of comments I wrote. 
Good work!

**Comment by ghost** [comment]
Thank you! We'll try to integrate in the changes you mentioned into our next few commits.
